File: lightbulb-recycleradapters/src/main/java/com/github/RooneyAndShadows/lightbulb/recycleradapters/EasyAdapterItemsComparator.java
package com.github.rooneyandshadows.lightbulb.recycleradapters;

public abstract class EasyAdapterItemsComparator<T extends EasyAdapterDataModel> {
    public abstract boolean compareItems(T oldItem, T newItem);

    public abstract boolean compareItemsContent(T oldItem, T newItem);

}

File: lightbulb-recycleradapters/src/main/java/com/github/rooneyandshadowss/lightbulb/recycleradapters/abstraction/EasyAdapterItemsComparator.kt
package com.github.rooneyandshadowss.lightbulb.recycleradapters.abstraction

abstract class EasyAdapterItemsComparator<T : EasyAdapterDataModel?> {
    abstract fun compareItems(oldItem: T, newItem: T): Boolean
    abstract fun compareItemsContent(oldItem: T, newItem: T): Boolean
}